sktree/tree/unsupervised/_unsup_criterion.pyx

```diff
@@ -398,3 +397,175 @@ cdef class TwoMeans(UnsupervisedCriterion):
+        2. Mean = mu
+        3. Vars = sig
+
+        impurity = N*log(2*sig)-N*log(mu)
+        """
+        cdef double mean
+        cdef double impurity
+        cdef SIZE_t n_node_samples = self.n_node_samples
+
+        # If calling without setting the
+        if self.Xf is None:
+            with gil:
+                raise MemoryError(
+                    'Xf has not been set yet, so one must call init_feature_vec.'
+                )
+
+        # first compute mean
+        mean = self.sum_total / self.weighted_n_node_samples
+
+        # then compute the variance of the cluster
+        sig = self.sum_of_squares(
+            self.start,
+            self.end,
+            mean
+        ) / self.weighted_n_node_samples
+
+        impurity = n_node_samples*log(2*sig/mean)
+
+        return impurity
+
+    cdef void children_impurity(
+        self,
+        double* impurity_left,
+        double* impurity_right
+    ) nogil:
+        """Evaluate the impurity in children nodes.
+
+        i.e. the impurity of the left child (sample_indices[start:pos]) and the
+        impurity the right child (sample_indices[pos:end]).
+
+        TODO: describe FastBIC
+
+        [Equation here]
+
+        - left_variance = left_weight * left_impurity / n_sample_of_left_child
+        - right_variance = right_weight * right_impurity / n_sample_of_right_child
+
+        Parameters
+        ----------
+        impurity_left : double pointer
+            The memory address to save the impurity of the left node
+        impurity_right : double pointer
+            The memory address to save the impurity of the right node
+        """
+        cdef SIZE_t pos = self.pos
+        cdef SIZE_t start = self.start
+        cdef SIZE_t end = self.end
+        cdef SIZE_t s_l
+        cdef SIZE_t s_r
+        cdef double p_l
+        cdef double p_r
+        cdef double mean_left
+        cdef double mean_right
+        cdef double sig_left
+        cdef double sig_right
+        cdef double left_term
+        cdef double right_term
+
+        # number of samples of left and right
+        s_l = pos - start
+        s_r = end - pos
+
+        # compute prior
+        p_l = s_l / self.n_node_samples
+        p_r = s_r / self.n_node_samples
+
+        # first compute mean of left and right
+        mean_left = self.sum_left / self.weighted_n_left
+        mean_right = self.sum_right / self.weighted_n_right
+
+        sig_left = self.sum_of_squares(
+            start,
+            pos,
+            mean_left
+        ) / self.weighted_n_left
+
+        sig_right = self.sum_of_squares(
+            pos,
+            end,
+            mean_right
+        ) / self.weighted_n_right
+
+        left_term = log(2*p_l*sig_left/mean_left)
+        right_term = log(2*p_r*sig_right/mean_right)
+
+        # set values at the address pointer
+        impurity_left[0] = s_l*left_term + s_r*right_term
+        impurity_right[0] = s_r*left_term + s_l*right_term
```

Comment:
@adam2392 is this heading towards the right direction?

Reply:
Can you add an in-line comment specifying the equation each one is computing just for ease of use?

Yes this is the right direction and looks correct to me.